src/java.base/share/classes/java/lang/Class.java

```diff
@@ -624,7 +624,15 @@ public static Class<?> forName(Module module, String name) {
      * @since Valhalla
      */
     @PreviewFeature(feature = PreviewFeature.Feature.VALUE_OBJECTS, reflective=true)
-    public native boolean isIdentity();
+    public boolean isIdentity() {
+        if (isPrimitive()) {
+            return false;
+        } else if (PreviewFeatures.isEnabled()) {
+           return isArray() || Modifier.isIdentity(modifiers);
+        } else {
+            return !isInterface();
+        }
```

Comment:
Isn't this going to take a performance hit?

Reply:
Yes this is better performance because C2 now doesn't have to intrinsify this native call.

Reply:
I'd go ahead and create the isIdentity field and move this code to the constructor.
The modifiers, component type and primitive flag are all available at once.